R/make_instrument_auto.R

```diff
@@ -149,7 +149,7 @@ fix_class_bug <- function(df) {
 
   # Applying the change class names to the exported REDCap data.
   for (i in 1:ncol(df)) {
-    class(df[, i]) <- classes2_ls[[i]]
+    class(df[[i]]) <- classes2_ls[[i]]
```

Comment:
rewrite it so it works with tibbles (which will return a single-column rectangle) _and_ works with data.frames (which will return a vector)

Reply:
It probably makes sense for an R package called *tidy*REDCap to work naturally with tibbles (since they're a big part of tidyverse)

Reply:
Gabriel, do you recall why we did this?  The original code is icky.  I don't know why this worked.  I am guessing that the right-hand-side is a list with a single item so it is being made into a vector which can be put into the left-hand-side.   I think Will's code makes sense but I would love to see this in a browser to understand what is going on.